ambari-server/src/main/java/org/apache/ambari/server/security/authorization/AmbariLdapAuthenticationProvider.java

```diff
@@ -264,7 +264,7 @@ private UserEntity getUserEntity(Authentication authentication) {
 
         if (!CollectionUtils.isEmpty(authenticationEntities)) {
           for (UserAuthenticationEntity entity : authenticationEntities) {
-            if (!StringUtils.isEmpty(entity.getAuthenticationKey())) {
+            if (StringUtils.isEmpty(entity.getAuthenticationKey())) {
```

Comment:
Just wanted to make sure this is what you intended - with no auth key, you trust that the user is correct?

Reply:
This is _now_ correct. I was fixing a bug.  A comment a few line up (probably hidden in this PR) reads:

> // If a user was not found with the exact authentication properties (LDAP/dn), look up the user
> // using the configured LDAP username attribute and ensure that user has an empty-keyed LDAP
> // authentication entity record.

This is the clause that determines if the key is empty.